### ISSUE TITLE
[feature request] nbviewer-like urls to access notebooks in GitHub repos and gists

### ISSUE THREAD

**Issue by vadimkantorov** [iodide-project/iodide#1822]
Putting iodide notebooks under version control (along with other data handling code) seems very reasonable (they are very diff-friendly!).

Proposal:
1) nbviewer-like urls to open notebooks stored in github repos / gists. In this scenario saving back isn't supported
2) (advanced) Committing back to a given single notebook file (Iodide already has GitHub authorization, and may request additional permissions, or the user can provide a personal access token in Iodide settings)
3) (god-mode) A fully-fledged file browser enabling choosing and editing different files from a GitHub repo (this would be a move into the JupyterLab UI direction). Also there exists a successful attempt of building [libgit2 for webassembly](https://github.com/libgit2/libgit2/pull/4400) if a local git client is considered.

The issue was already discussed in https://github.com/iodide-project/iodide/issues/361, and it seems to have been met enthusiastically. I couldn't understand what's the team's current view of this feature. Happy to close this one if tracking in the old issue is preferred.

**Comment by bcolloran** [comment]
@vadimkantorov we decided to pull back from this a while ago because some of the workflows we want to enable (like instant automatic saving) was going to be hard/impossible to implement using github as our data backend, as well as finding git to be hard to understand for beginners and frankly overkill complicated for a lot of data/science work. since our team is too small to build and maintain both backends, we needed to pull back. Unfortunately, I don't think we'll be able to prioritize this any time soon, because we want to focus on making our own light weight text/version handling tools better (for example, we plan on migrating to the Monaco editor, which has an excellent diff viewer/editor that will go nicely with our versioning system).

Overall, I'd say that our current view is that: this is not a high priority and we're very resource constrained so we probably won't get to it in the foreseeable future, but we'd accept a PR implementing export/import from GH. If you are interested in working on it, we can discuss it in more detail. If not, that's fine too -- we can leave this issue to track it for when we have some time to work on it.

thank you for filing this issue, and thanks for your support! :-)
